lib/driver.js

```diff
@@ -371,20 +371,24 @@ class XCUITestDriver extends BaseDriver {
         await iosSettings.setLocaleAndPreferences(sim, this.opts, this.isSafari());
       });
 
-      await this.startSim();
-
       if (this.opts.customSSLCert) {
-        if (await hasSSLCert(this.opts.customSSLCert, this.opts.udid)) {
-          log.info(`SSL cert '${_.truncate(this.opts.customSSLCert, {length: 20})}' already installed`);
-        } else {
-          log.info(`Installing ssl cert '${_.truncate(this.opts.customSSLCert, {length: 20})}'`);
-          await shutdownSimulator(this.opts.device);
-          await installSSLCert(this.opts.customSSLCert, this.opts.udid);
-          log.info(`Restarting Simulator so that SSL certificate installation takes effect`);
-          await this.startSim();
-          this.logEvent('customCertInstalled');
+        try {
+          await this.opts.device.simctl.addRootCertificate(this.opts.customSSLCert, {raw: true});
+        } catch (ign) {
+          if (await hasSSLCert(this.opts.customSSLCert, this.opts.udid)) {
+            log.info(`SSL cert '${_.truncate(this.opts.customSSLCert, {length: 20})}' already installed`);
```

Comment:
we're doing this truncating twice, does it make sense to only do it once?

Reply:
changed